sdk/azidentity/managed_identity_credential.go

```diff
@@ -64,6 +65,14 @@ func NewManagedIdentityCredential(clientID string, options *ManagedIdentityCrede
 // scopes: The list of scopes for which the token will have access.
 // Returns an AccessToken which can be used to authenticate service client calls.
 func (c *ManagedIdentityCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	if opts.Scopes == nil {
+		return nil, errors.New("must specify a resource in order to authenticate")
```

Comment:
Shouldn't this, and below, return a `CredentialUnavailableError`?

Reply:
So I did think about this. I believe at one point we said that CredentialUnavailableError would only be called when the conditions to create the credential are not available (i.e. in the ctor). And AuthenticationFailedErrors would be exactly for when we attempt to authenticate but cannot do so. So I'm more tempted to go with AuthenticationFailedError since this is in the call to GetToken(). What do you think?

Reply:
AuthenticationFailedError is fine (frankly either one would work, I think the important part is it's not a naked error).

Reply:
Done, also added logging.